src/offline/syncUpdatesWithBackend.ts

```diff
@@ -1,176 +1,348 @@
         //Syncronize updates for the specific entity
 
         //Find all updates for the current entity
         const updatesForEntity = offlineUpdates.filter(
-            (offlineUpdate) => offlineUpdate.entityid == entityIdToUpdate
+            (offlineUpdate) =>
+                offlineUpdate.entityId == entityToUpdate.id &&
+                offlineUpdate.entityType == entityToUpdate.type
         );
 
-        let newEntityId;
+        console.log(
+            'Skal gjøre oppdateringer for entity. ',
+            entityToUpdate,
+            updatesForEntity
+        );
 
+        //Perform updates for the given entity.
+        //If an error occurs, further synchronization of this entity will be stopped and
+        //the updates will be marked with 'error'.
+        //If the update is already set to synchronized, or an error code is set, it will be skipped.
+        let newEntityId;
         for (const offlineUpdate of updatesForEntity) {
-            if (!offlineUpdate.syncronized) {
+            if (
+                offlineUpdate.syncStatus == SyncStatus.NOT_SYNCHRONIZED &&
+                !offlineUpdate.errorCode
+            ) {
                 if (newEntityId) {
-                    //A previous POST for the entity returned an new entityId generated by backend.
+                    //A previous POST for the entity returned a new entityId generated by backend.
                     //The id for the update request for this entity must be updated.
                     updateEntityId(newEntityId, offlineUpdate);
                 }
-                const id = await syncronizeOfflineUpdate(offlineUpdate);
+                const id = await synchronizeOfflineUpdate(offlineUpdate, api);
                 if (id) {
                     //If id is returned it means that backend has created a new ID, that must be used for subsequent updates for this entity.
                     newEntityId = id.Id;
                 }
-            } else {
+            } else if (offlineUpdate.syncStatus == SyncStatus.SYNCHRONIZED) {
                 console.log(
                     'Offline update already syncronized for entity ' +
-                        offlineUpdate.entityid
+                        offlineUpdate.entityId
+                );
+            } else if (offlineUpdate.syncStatus == SyncStatus.ERROR) {
+                console.log(
+                    `Offline update already set to error-status:  ${offlineUpdate.errorCode} - ${offlineUpdate.errorMessage} `
+                );
+                break; //If an error occured we will skip further updates on this entity.
+            } else {
+                console.log(
+                    'The offline update does not have the corret synchronization status. ',
+                    offlineUpdate.syncStatus
+                );
+            }
+        }
+    }
+
+    await reportErrors(currentPlant, currentProject, api);
+    await api.putOfflineScopeSynchronized(currentPlant, currentProject);
```

Comment:
We can't call both sync success and sync failed (as done in reportErrors), can we?

Reply:
Yes we can. We will set to synchronized even if there are errors